shared/naturalcrit/markdown.js

```diff
@@ -7,14 +7,44 @@ const { gfmHeadingId: MarkedGFMHeadingId } = require('marked-gfm-heading-id');
 const renderer = new Marked.Renderer();
 
 //Processes the markdown within an HTML block if it's just a class-wrapper
-renderer.html = function (html) {
-	if(_.startsWith(_.trim(html), '<div') && _.endsWith(_.trim(html), '</div>')){
-		const openTag = html.substring(0, html.indexOf('>')+1);
-		html = html.substring(html.indexOf('>')+1);
-		html = html.substring(0, html.lastIndexOf('</div>'));
-		return `${openTag} ${Marked.parse(html)} </div>`;
+// renderer.html = function (html) {
+// 	if(_.startsWith(_.trim(html), '<div') && _.endsWith(_.trim(html), '</div>')){
+// 		const openTag = html.substring(0, html.indexOf('>')+1);
+// 		html = html.substring(html.indexOf('>')+1);
+// 		html = html.substring(0, html.lastIndexOf('</div>'));
+// 		return `${openTag} ${html} </div>`;
+// 	}
+// 	return html;
+// };
+
+//Processes the markdown within an HTML block if it's just a class-wrapper
+const htmlDiv = {
+	name  : 'htmlDiv',
+	level : 'block',
+	tokenizer(src, tokens) {
+		let html = src;
+		if(_.startsWith(_.trim(html), '<div') && _.endsWith(_.trim(html), '</div>')){
+			const openTag = html.substring(0, html.indexOf('>')+1);
+			html = html.substring(html.indexOf('>')+1);
+			html = html.substring(0, html.lastIndexOf('</div>'));
+
+			const item = {                                                  // Token to generate
+				type    : 'htmlDiv',                                        // Should match "name" above
+				raw     : src.substring(0, src.indexOf('</div>')+6),        // Text to consume from the source
+				text    : html,                                             // Additional custom properties
+				openTag : openTag,
+				tokens  : []
+			};
+
+			this.lexer.blockTokens(html, item.tokens);
+
+			return item;
+		}
+		return false;
+	},
+	renderer(token) {
+		return `${token.openTag} ${this.parser.parse(token.tokens)} </div>`;
 	}
```

Comment:
This does not work. Try testing this text. Both should have functional markdown:

```
<div>

Will this *render?*

</div>

<div>
Will this *render?*
</div>
```

The problem is most of the logic in the original code here is applied to the Renderer, but this change has put it inside the Tokenizer, so the tokens being generated aren't correct.

I agree this bit of code should be made into correct "Markdown Extension" format. However, I think it's outside the scope of this PR and will need more work that complicates the `\page` token stuff. Can you remove it from here and put into a separate PR we can better test in isolation?

Reply:
Oh, I remember this now! Because this was using a separate instance of the renderer, it was generating a `pageBlock` to contain the HTML div; that's why I ended up including this change here.

Reply:
Something is definitely screwy, though:

![image](https://github.com/naturalcrit/homebrewery/assets/19826659/8a0ab2d4-6023-49bc-aa22-0806b6452747)

Reply:
> this was using a separate instance of the renderer

I don't see where a separate instance is being used? We still have `Marked.use({ renderer: renderer, mangle: false });` where the original HTML handling code is being applied to the same instance.

Reply:
I think we cut this out, make it it's own PR, get it merged
Then come back for the larger `\page` token PR

Reply:
> I don't see where a separate instance is being used?

I think it was the `Marked.parse()` call; although I could be wrong about that.

Reply:
With this change reverted, I'm not seeing the behaviour that I was seeing earlier - I probably blamed this particular chunk of code for something unrelated.